src/cache_api.rs

```diff
@@ -1,14 +1,27 @@
 use tokio::task::JoinHandle;
 use std::hash::Hash;
 use futures::Future;
+use thiserror::Error;
 use crate::internal_cache::{CacheAction, InternalCacheStore, CacheMessage};
 use crate::backing::{CacheBacking, HashMapBacking};
+use std::fmt::Debug;
 
-#[derive(Debug, Clone)]
-pub struct CacheLoadingError {
-    pub reason_phrase: String,
-    pub loading_error: Option<LoadingError>,
-    // todo: nested errors
+#[derive(Error, Debug)]
+pub enum CacheLoadingError<E: Debug> {
+    #[error("An error occurred when trying to submit the cache request")]
+    TokioMpscSendError(),
+    #[error("An error occurred when trying to join the result future")]
+    FutureJoinError(tokio::task::JoinError),
```

Comment:
Consider using `#[from]` for each of these to automatically generate `Into` implementations: 
```suggestion
    FutureJoinError(#[from] tokio::task::JoinError),
```

Reply:
Your suggested change doesn't compile.
 I'm not quite sure, but it seems the derive generates some code which clashes with the code generated from the #[from].
 I'll check if its a simple thing, or if I just implement the Into traits myself.

Reply:
Interesting- if you post the compilation error, I can suggest what may be breaking.